packages/roosterjs-editor-dom/test/selections/deleteSelectedContentTest.ts

```diff
@@ -0,0 +1,141 @@
+    });
+
+    it('Text input', () => {
+        runTest('test<!--{"start":[0,1],"end":[0,3]}-->', 'tt<!--{"start":[0,1],"end":[0,1]}-->');
+    });
+
+    it('Two text node', () => {
+        runTest(
+            'test1<br>test2<!--{"start":[0,2],"end":[2,2]}-->',
+            'test2<!--{"start":[0,2],"end":[0,2]}-->'
+        );
+    });
+
+    it('Two DIV node', () => {
+        runTest(
+            '<div>test1</div><div>test2</div><!--{"start":[0,0,2],"end":[1,0,2]}-->',
+            '<div>test2</div><!--{"start":[0,0,2],"end":[0,0,2]}-->'
+        );
+    });
+
+    it('Two DIV node, select to next line', () => {
+        runTest(
+            '<div>test1</div><div>test2</div><!--{"start":[0,0,2],"end":[1,0]}-->',
+            '<div>tetest2</div><!--{"start":[0,0,2],"end":[0,0,2]}-->'
+        );
+    });
+
+    it('Simple list', () => {
+        runTest(
+            '<ol><li>line1</li><li>line2</li><li>line3</li></ol><!--{"start":[0,0,0,2],"end":[0,1,0,2]}-->',
+            '<ol><li>line2</li><li>line3</li></ol><!--{"start":[0,0,0,2],"end":[0,0,0,2]}-->'
+        );
+    });
+
+    it('Nested list', () => {
+        runTest(
+            '<ol><li>line1</li><ol><li>line2</li><li>line3</li></ol><li>line4</li></ol><!--{"start":[0,0,0,2],"end":[0,1,0,0,2]}-->',
+            '<ol><li>line2</li><ol><li>line3</li></ol><li>line4</li></ol><!--{"start":[0,0,0,2],"end":[0,0,0,2]}-->'
+        );
+    });
+
+    it('Nested list and text', () => {
+        runTest(
+            '<ol><li>line1</li><ol><li>line2</li><li>line3</li></ol><li>line4</li></ol>line5<br>line6<!--{"start":[0,1,1,0,2],"end":[1,2]}-->',
+            '<ol><li>line1</li><ol><li>line2</li><li>li<span>ne5<br></span></li></ol></ol>line6<!--{"start":[0,1,1,0,2],"end":[0,1,1,0,2]}-->'
+        );
+    });
+
+    it('Nested list and text and nested list', () => {
+        runTest(
+            '<ol><li>line1</li><ol><li>line2</li><li>line3</li></ol><li>line4</li></ol>line5<br>line6<ol><li>line7</li><ol><li>line8</li><li>line9</li></ol><li>line10</li></ol><!--{"start":[0,1,1,0,2],"end":[4,1,0,0,2]}-->',
+            '<ol><li>line1</li><ol><li>line2</li><li>line8</li></ol></ol><ol><ol><li>line9</li></ol><li>line10</li></ol><!--{"start":[0,1,1,0,2],"end":[0,1,1,0,2]}-->'
+        );
+    });
+
+    it('Whole talbe 1', () => {
+        runTest(
+            'aa<table><tbody><tr><td>line1</td><td>line2</td></tr><tr><td>line3</td><td>line4</td></tr></tbody></table>bb<!--{"start":[0,2],"end":[2,0]}-->',
+            'aabb<!--{"start":[0,2],"end":[0,2]}-->'
+        );
+    });
+
+    it('Whole talbe 2', () => {
+        // TODO: the result contains separated continuous text object at root
+        // Selection path gives wrong result. Need to revisit here
+        runTest(
+            'aa<table><tbody><tr><td>line1</td><td>line2</td></tr><tr><td>line3</td><td>line4</td></tr></tbody></table>bb<!--{"start":[1],"end":[2]}-->',
+            'aabb<!--{"start":[],"end":[]}-->'
```

Comment:
So from what I can tell here... the issue seems to be that on deleteSelectedContent the startContainer is the text node "line1" which is text but has the parent "td". Then when ensureBeforeAndAfter node, it creates a new node with "" but which also has parentElement td. So when creating the new Position to select, the nodeBefore (which is the empty text with parent td) will try to findClosestElementAncestor, but the parent node is td.

But I wonder if this specific selection path scenario can happen outside of a test. When doing it manually I can cut the table and the selection after works.

Reply:
Not quite understand what you mean here. You can try the selection path here:

![image](https://user-images.githubusercontent.com/23065085/94061129-d2ff6d00-fd99-11ea-9307-cc6a2b86e707.png)

You can put some text, make a selection and click "Take snapshot" to create a snapshot with selection path, modify it then click "restore snapshot" to apply it to editor.

Reply:
Got it, thanks. I was just wondering if we needed to worry about fixing that specific scenario because I didn't know if our users could get into that state but I see now that there are ways to set a very specific selectionPath.

Reply:
This has been there for long time and it should not be easy to hit.